decidim-core/app/helpers/decidim/external_domain_helper.rb

```diff
@@ -4,10 +4,21 @@ module Decidim
   module ExternalDomainHelper
     # Redesign: deprecated once the redesign has been completed
     def highlight_domain
+      highlighted_domain = [
+        external_url.host,
+        (external_url.port && [80, 443].include?(external_url.port) ? "" : ":#{external_url.port}")
+      ].join
+
+      path = [
+        external_url.path,
+        (external_url.query ? "?#{external_url.query}" : ""),
+        (external_url.fragment ? "##{external_url.fragment}" : "")
+      ].join
+
       tag.div do
-        content_tag(:span, "#{@url_parts[:protocol]}//") +
-          content_tag(:span, @url_parts[:domain], class: "alert") +
-          content_tag(:span, @url_parts[:path])
+        content_tag(:span, "#{external_url.scheme}://") +
+          content_tag(:span, highlighted_domain, class: "alert") +
```

Comment:
Here you also need to add the same class as there was before in the modal view:
```suggestion
          content_tag(:span, highlighted_domain, class: "text-alert") +
```

Otherwise the domain is not highlighted under redesign.

Before (this is what it should look like):
![Link highlight working](https://user-images.githubusercontent.com/864340/228846024-86dea9ce-d1df-46cf-ab59-68e985ffb77a.png)

After this class change (this is incorrect):
![Link highlight broken](https://user-images.githubusercontent.com/864340/228846216-09cc7af6-9095-4ed1-9a3e-39e38be3d580.png)

Note that if we want to backport these changes to 0.27 or before, we need to change it back to `alert` for the highlight to work correctly under the legacy design.